### ISSUE TITLE
Error on Location.startLocationUpdatesAsync

### ISSUE THREAD

**Issue by Javierko** [expo/expo#11330]
### Summary of Issue 
Impossible to start background location updating. Getting this error:
`Unhandled promise rejection: Error: Not authorized to use background location services.`

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
```
Expo CLI 4.0.12 environment info:
    System:
      OS: Windows 10 10.0.19041
    Binaries:
      Node: 14.15.1 - E:\PHP\laragon\bin\nodejs\node-v14\node.EXE
      Yarn: 1.21.1 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 6.14.8 - E:\PHP\laragon\bin\nodejs\node-v14\npm.CMD
    IDEs:
      Android Studio: Version  4.0.0.0 AI-193.6911.18.40.6626763
    npmPackages:
      expo: ^40.0.0 => 40.0.0
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: https://github.com/expo/react-native/archive/sdk-40.0.0.tar.gz => 0.63.2
      react-native-web: ~0.13.12 => 0.13.18
    Expo Workflow: managed
```

### Reproducible Demo
```
const LOCATION_TRACKING = 'background-location-tracking';

export default class App extends Component
{
    componentDidMount() {
        this.locationTracking();
    }

    locationTracking = async () => {
        const { status } = await Location.requestPermissionsAsync();

        if (status === 'granted') {
            let location = await Location.startLocationUpdatesAsync(LOCATION_TRACKING, {
                accuracy: Location.Accuracy.BestForNavigation,
                showsBackgroundLocationIndicator: true,
                timeInterval: 30000,
                activityType: Location.ActivityType.AutomotiveNavigation,
                distanceInterval: 1
            });
        }
    }
}

TaskManager.defineTask(LOCATION_TRACKING, async ({ data, error }) => {
    if (error) {
        console.log('LOCATION_TRACKING task ERROR:', error);

        return;
    }

    if(data) {
        const { locations } = data;

        console.log('send to api');
    }
});
```

 
### Steps to Reproduce
Just use to code.

### Expected Behavior vs Actual Behavior
To not get the error. When trying to debug `status` i get `granted`.

**Comment by sumit1317** [comment]
Facing the same problem ...

**Comment by cruzach** [comment]
As stated in the [changelog](https://github.com/expo/expo/blob/master/CHANGELOG.md#-breaking-changes-1) and our [SDK 40 blog post](https://blog.expo.io/expo-sdk-40-is-now-available-d4d73e67da33), Android background location permissions are now opt-in. here's the relevant PR for more info- https://github.com/expo/expo/pull/10989

**Comment by JosephLep** [comment]
I got approved my android for this - yet I am still getting the error above.  I have added:

`"android": {
      "useNextNotificationsApi": true,
      "permissions": [ "ACCESS_BACKGROUND_LOCATION" ],.....`

Am I missing something else here?

**Comment by sumit1317** [comment]
@JosephLep , how you got your app approved from Google, can you help me through ?
About iOS, for me it works on Expo Simulator on Mac, but not on Expo inside iOS phone, have you done anything for it ?

**Comment by rpstew** [comment]
I am seeing this same issue on my android phone:

[Unhandled promise rejection: Error: Not authorized to use background location services.]

Location updates work when the app is in the foreground, but I get that error when the app is put into the background.

**Comment by JosephLep** [comment]
@sumit1317 What I had to do was record a video that had the android emulator in view as well as the settings for the emulator.  I had to minimize my app, move location using the GPS/Location stuff in the emulator settings, have some output viewable for them in the video, relaunch the app (in my case) to show them what if any changes had happened while backgrounded.  Took me a few tries of submitting different videos - but try to be as detailed as possible.

**Comment by JosephLep** [comment]
@rpstew Even in the foreground when I try to run geofence methods I get that same error.

**Comment by sumit1317** [comment]
@JosephLep , I understand that when the app has to loaded into any Play Store, then relevant permission will be required (which you have mentioned above is in the form of a video), but my concern is do we need permission for developing the app, and running it in our own machine/ inside expo client ?

**Comment by JosephLep** [comment]
@sumit1317 with very unclear direction from Expo - that is a very good question.  It seems like we do have to get this permission first before dev.  I have no idea how this works still.  Like I mentioned above - before I could not get it to work and even after getting approved from Google - I still can not get it to work inside of the expo client.

**Comment by byCedric** [comment]
Hi people, I wanted to give a quick update on this issue for all of you. It's unfortunate that Google is rapidly changing the permissions around locations. Thanks to @peterdn, we have a fix ready for Android <9. Android 10 works as expected, but for Android 11 we need to do more testing and possibly refactoring. Here is why:

- **Android <9**
On this version of Android, background permission was similar to foreground permission. When asking permission for `ACCESS_COARSE_LOCATION` and/or `ACCESS_FINE_LOCATION`, you already received access to also fetch the location in the background. `ACCESS_BACKGROUND_LOCATION` doesn't exist in this version and should not be requested to users ([see this](https://developer.android.com/reference/android/Manifest.permission#ACCESS_BACKGROUND_LOCATION), added in API 29/Android 10). That's [why this is excluded](https://github.com/expo/expo/blob/master/packages/expo-permissions/android/src/main/java/expo/modules/permissions/requesters/LocationRequester.kt#L24-L26) when requesting permission for `LOCATION` in Android, through `Permissions.askAsync(Permissions.LOCATION)`.

- **Android 10**
This version of Android includes the new `ACCESS_BACKGROUND_LOCATION`, which used to be auto-requested in Expo. Because Google's policy is now more "aggressive" against apps that request this permission (because of the additional audit), we made it an opt-in. Right now, the code is handling this properly and should do so after we deploy the latest fix.

- **Android 11**
The latest version of Android includes another change related to the location. We can't request both "foreground" and "background" location at the same time. It also "removes" the option for users to "allow location access all the time" from the permission dialog and puts this under the settings (you can see the screens below). This is something to be aware of when creating your app.

I'll create an example app that uses the background location permission and link the source here. That app should be published on the Play store, but I can't estimate right now how long that will take. Once I have more to share, I will keep you updated.

**Android 11 Location permission screens**
permission dialog | location setting
--- | ---
![android-11-location-dialog](https://user-images.githubusercontent.com/1203991/102638340-9f916c00-4157-11eb-8398-7844a4b30f4f.png) | ![android-11-location-settings](https://user-images.githubusercontent.com/1203991/102638341-a0c29900-4157-11eb-9406-faf84c3a5c30.png)

**Comment by byCedric** [comment]
Another quick update on this one!

- As promised, I [created a sample app](https://github.com/bycedric/office-marathon#readme) that I will publish to the Play store once the standalone builds are updated too. PR is out, but staging is not yet updated.
- Staging just got updated with Expo Client `2.18.5`, meaning you can try this version of Expo Client on your phone. Set envvar `EXPO_STAGING=1` and install it on your emulator or device. On windows you can use `npx cross-env EXP...` as prefix. Once this version is working as intended, we can deploy it to production.
```
$ EXPO_STAGING=1 expo client:install:android
```

Once everything is deployed to production I'll let you know and close this issue. Feel free to ping me if background location is still not working as intended on Expo Client `2.18.5`.

Hope this helps!

**Comment by sumit1317** [comment]
Replicated the app from Git... but still receiving the error messages.

*iOS* *Device*-

*[Unhandled promise rejection: Error: Background Location has not been
configured. To enable it, add `location` to `UIBackgroundModes` in
Info.plist file.]*


*Android* *Device**- **[Unhandled promise rejection: Error: Not authorized
to use background location services.]*


Running perfectly on iOS Simulator.


Did the following to run your code.

1. Downloaded .zip from Git and Extracted into a folder

2. Ran "npm install"

3. Ran "expo start"

4. Than on iOS Simulator - it worked fine

5. Ran your app inside Expo App on iOS phone - gave the error message as
mentioned above

6. Ran your app inside Expo App in Android phone - gave the error message
as mentioned above



Regards,
*Sumit S Chadha*

**Comment by rwanito** [comment]
Hi !

I have the same issue.
Can we have the certitude that we get the background permissions ?
And, maybe can we do : IntentLauncher to go in locations settings ?

Thanks !

**Comment by byCedric** [comment]
Last update 😄 

### Deployment
- Expo Client `2.18.5` is released to production, you can install it through `$ expo client:install:android`.
- Standalone builds will also include this fix, go a head and start your new build through `$ expo build:android`.

### How to use it
- For managed users, you don't have to do anything special (no node module updates, just use Expo Client `2.18.5+` and build a new standalone app).
- For bare users, we will push a new update for `expo-location` with the native code changes ASAP. Unfortunately, we have to check some differences out first. Meanwhile, you can use something like [`patch-package`](https://www.npmjs.com/package/patch-package) to [include this change](https://github.com/expo/expo/commit/304cd013d8337f97012c76d2ffb09601f508339e)

### Sample app
- App has been submitted to Google Play, once that's approved I'll [merge the PR](https://github.com/byCedric/office-marathon/pull/1) to link to it directly. For people who can't wait, you can [download the APK from here](https://exp-shell-app-assets.s3.us-west-1.amazonaws.com/android/%40bycedric/office-marathon-bd3b9eb64880468194023cb798dba4fa-signed.apk).

### Android support
- Android <9 and Android 10 should work without issues now.
- Android 11 will require more changes to make it fully work with [Google's new stricter privacy changes](https://developer.android.com/about/versions/11/privacy/location). More on that soon!

Happy holidays!

**Comment by superyarik** [comment]
@byCedric, is this fix suitable for latest 40 SDK? or for all supported at this moment SDKs?

**Comment by rwanito** [comment]
@byCedric Do you have any news about that ?
I'm stuck on Android and I don't want to temporarily rollback to Expo 39.

Thanks for your work !

**Comment by superyarik** [comment]
we have to stay on 39 too bacause this

**Comment by MaRaSu** [comment]
@rwanito, @superyarik and anyone else, there is new update on this available in #11854.
